### PR TITLE
Move aspect ratio computing to model

### PR DESCRIPTION
**Pull request by Jupi007**
Fix my mistake in #408

### CODE REVIEW

File: lib/view/pages/displays/displays_configuration.dart [hunk 1]
```diff
@@ -94,6 +94,14 @@ class DisplayMonitorConfiguration extends Equatable {
...
   String get resolution => _resolution;
 
+  double get aspectRatio {
+    final resolution = _resolution.split('x');
+    double aspectRatio =
+        (int.parse(resolution.first) / int.parse(resolution.last));
+
+    return aspectRatio;
+  }
+
```
Comment (Feichtmeier) [inline lib/view/pages/displays/displays_configuration.dart:104]: Almost :P

Make the service a parameter of the wallpaper model and inject it inside the create method of the wallpaper page

Then create the aspect ratio getter in the model 

Then call

model.adpectRatio

**Review by Feichtmeier — CHANGES_REQUESTED** [review]

File: lib/view/pages/displays/displays_configuration.dart (hunk 1 above)
Comment (Jupi007) [reply lib/view/pages/displays/displays_configuration.dart:104]: Okay :D
Sounds better indeed :+1:

Comment (Jupi007) [reply lib/view/pages/displays/displays_configuration.dart]: I moved `aspectRatio` getter to the wallpaper model.
Now it grabs the primary screen ratio, and it also watches screen configuration changes :)

File: lib/view/pages/wallpaper/wallpaper_model.dart [hunk 2]
```diff
@@ -34,9 +37,16 @@ class WallpaperModel extends SafeChangeNotifier {
...
-  WallpaperModel(SettingsService service)
-      : _wallpaperSettings = service.lookup(schemaBackground) {
+  DisplaysConfiguration? _displaysConfiguration;
+
+  WallpaperModel(
+    SettingsService wallpaperService,
+    DisplayService displayService,
+  ) : _wallpaperSettings = wallpaperService.lookup(schemaBackground) {
     _wallpaperSettings?.addListener(notifyListeners);
+    displayService.monitorStateStream.listen((configuration) {
```
Comment (Feichtmeier) [inline lib/view/pages/wallpaper/wallpaper_model.dart:47]: You should save the steam sub as a nullable ref and then cancel it in the model dispose

**Review by Feichtmeier — CHANGES_REQUESTED** [review]

File: lib/view/pages/wallpaper/wallpaper_model.dart (hunk 2 above)
Comment (Jupi007) [reply lib/view/pages/wallpaper/wallpaper_model.dart]: You're right, I just fixed it 🙂

In fact, I looked at `DisplaysModel` but I didn't see any disposal 🤔
Is it normal btw?

**Review by Feichtmeier — APPROVED** [review]

File: lib/view/pages/wallpaper/wallpaper_model.dart (hunk 2 above)
Comment (Feichtmeier) [reply lib/view/pages/wallpaper/wallpaper_model.dart]: Thanks! 
Github displaying review comments if weird xD

Yes the safe change notifiers don't need a dedicated dispose call if no other special resource is being used like streams

**Pull request merged**